src/runtime_src/core/pcie/noop/shim.cpp

```diff
@@ -338,6 +518,32 @@ get_shim_object(xclDeviceHandle handle)
 
 } // namespace
 
+namespace userpf {
+
+xrt_core::query::kds_cu_info::result_type
+kds_cu_info(const xrt_core::device* device)
+{
+  auto id = device->get_device_id();
+  if (s_devices.size() < id)
```

Comment:
Unfortunately, I don't know what is happening here with regards to device id allocation / deallocation.  

If it is possible for a random device to be "deallocated" then the s_devices collection size might not be the be best way to determine if the device's ID is correct.  

If the assumption is that once a device has been allocated, the class will only be deleted when the driver is shut down, then this code works.  

Again, don't know.  For me, the this code check for a validate ID just doesn't look right.

Reply:
The static list of devices is allocated and populated by xclOpen.  It is indexed by the device id, so this is really an internal error.  Anyways, this noop code is for quick and dirty debugging of user space pretending some device.